### PR TITLE
MCR-2150 URN cannot be registered

### PR DESCRIPTION
**Pull request by chiliego**
[Link to jira](https://mycore.atlassian.net/browse/MCR-).

### CODE REVIEW

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java [hunk 1]
```diff
@@ -269,6 +273,36 @@ protected void update(MCRDNBURN identifier, MCRBase obj, String additional)
...
 
+    @Override
+    public void updateFlag(MCRObjectID id, String additional, MCRPI mcrpi) {
+        MCRBase obj = MCRMetadataManager.retrieve(id);
+        MCRObjectService service = obj.getService();
+        ArrayList<String> flags = service.getFlags(MCRPIService.PI_FLAG);
+        Gson gson = getGson();
+
+        //just update flag for derivate, where additional is ""
+        if(additional.equals("")){
```
Comment (adlerre) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4992058)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4992058&bid=16478823)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 909b2e355db75faf67c08ee557e452ea4e60a0fb! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5242-pull_request-911

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4992058&bid=16478823)

File: mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java (hunk 1 above)
Comment (adlerre) [inline mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4992058)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-pi/src/main/java/org/mycore/pi/urn/rest/MCRURNGranularRESTService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4992058&bid=16478823)

**Review by Possommi — APPROVED** [review]

**Pull request merged**